tss-esapi/src/abstraction/transient/key_attestation.rs

```diff
@@ -0,0 +1,208 @@
+// Copyright 2021 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use super::{ObjectWrapper, TransientKeyContext};
+use crate::{
+    abstraction::ek,
+    constants::SessionType,
+    handles::{AuthHandle, KeyHandle, SessionHandle},
+    interface_types::{
+        algorithm::{AsymmetricAlgorithm, HashingAlgorithm},
+        session_handles::{AuthSession, PolicySession},
+    },
+    structures::{EncryptedSecret, IDObject, SymmetricDefinition},
+    tss2_esys::{Tss2_MU_TPMT_PUBLIC_Marshal, TPM2B_PUBLIC, TPMT_PUBLIC},
+    utils::PublicKey,
+    Error, Result, WrapperErrorKind,
+};
+use log::error;
+use std::convert::{TryFrom, TryInto};
+
+#[derive(Debug)]
+/// Wrapper for the parameters needed by MakeCredential
+pub struct MakeCredParams {
+    /// TPM name of the object
+    pub name: Vec<u8>,
+    /// Encoding of the public parameters of the object whose name
+    /// will be included in the credential computations
+    pub public: Vec<u8>,
+    /// Public part of the key used to protect the credential
+    pub attesting_key_pub: PublicKey,
+}
+
+impl TransientKeyContext {
+    /// Get the data required to perform a MakeCredential
+    ///
+    /// # Parameters
+    ///
+    /// * `object` - the object whose TPM name will be included in
+    /// the credential
+    /// * `key` - the key to be used to encrypt the secret that wraps
+    /// the credential
+    ///
+    /// **Note**: If no `key` is given, the default Endorsement Key
+    /// will be used.  
+    pub fn get_make_cred_params(
+        &mut self,
+        object: ObjectWrapper,
+        key: Option<ObjectWrapper>,
+    ) -> Result<MakeCredParams> {
+        let object_handle = self.load_key(object.params, object.material, None)?;
+        let (object_public, object_name, _) =
+            self.context.read_public(object_handle).or_else(|e| {
+                self.context.flush_context(object_handle.into())?;
+                Err(e)
+            })?;
+        self.context.flush_context(object_handle.into())?;
+
+        // Name of objects is derived from their publicArea, i.e. the marshaled TPMT_PUBLIC
+        let public = TPM2B_PUBLIC::from(object_public);
+        let mut pub_buf = [0u8; std::mem::size_of::<TPMT_PUBLIC>()];
+        let mut offset = 0;
+        let result = unsafe {
+            Tss2_MU_TPMT_PUBLIC_Marshal(
```

Comment:
Not sure if this was mentioned elsewhere but this looks like a good idea to have this function exposed for other code as well. Would it make sense to move it to the public wrapper or do you think it's not worth it?

Reply:
I'll raise an issue for it and we can figure it out independently of this PR, but yeah, I think it would be a good addition to the API, and it won't break the API either.

Reply:
Done, #291